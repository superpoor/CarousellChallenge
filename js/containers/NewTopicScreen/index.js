import React from 'react'
import {
  View, 
  Text,
  Button,
  Platform,
  TextInput,
  StyleSheet,
  KeyboardAvoidingView
} from 'react-native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Loading from '../../components/loading'
import { addNewTopic } from '../../actions/topics'


@connect(
  state => ({
    isAddingNewTopic: state.topics.isAddingNewTopic
  }),
  dispatch => bindActionCreators({
    addNewTopic
  }, dispatch)
)
class NewTopicScreen extends React.PureComponent {

  static propTypes = {
    isAddingNewTopic: PropTypes.bool.isRequired,
    addNewTopic: PropTypes.func.isRequired,
  };

  static navigationOptions = ({ navigation }) => {
    const { params = {} } = navigation.state;

    return {
      headerTitle: "New Topic",
      headerRight: (
        <Button  
          title="SUBMIT"
          onPress={params.handleSubmit ? params.handleSubmit : (() => null) }
        />
      )
    }
  };

  state = {
    inputValue: ''
  };

  render() {
    const { isAddingNewTopic } = this.props;
    const { inputValue } = this.state;

    if (isAddingNewTopic) return <Loading />
    
    return (
      <KeyboardAvoidingView 
        behavior= {"padding"}
        keyboardVerticalOffset={Platform.OS == 'ios' ? 64 : 80}
        style={styles.container}>
        <TextInput 
          style={styles.textInput}
          placeholder="Type your topic here"
          multiline={true}
          underlineColorAndroid="transparent"
          onChangeText={(text) => { if (text.length <= 255) this.setState({inputValue: text}) }}
          value={inputValue}
        />
        <Text style={styles.charLeft}>{255 - inputValue.length} characters left</Text>
      </KeyboardAvoidingView>
    )
  }

  componentDidMount() {
    this.props.navigation.setParams({ handleSubmit: this._handleSubmit });
  }

  _handleSubmit = () => {
    this.props.addNewTopic(this.state.inputValue);
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  textInput: {
    flex: 1,
    fontSize: 16,
    padding: 20,
    marginVertical: 20,
    textAlignVertical: 'top',
    color: '#545454'
  },
  charLeft: {
    color: '#747474',
    fontSize: 14,
    paddingRight: 10,
    paddingBottom: 10,
    textAlign: 'right'
  }
})

export default NewTopicScreen;
