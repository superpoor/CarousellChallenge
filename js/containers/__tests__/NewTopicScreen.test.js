import React from 'react';
import renderer from 'react-test-renderer';

import { NewTopicScreen } from '../NewTopicScreen'

describe('Test NewTopicScreen component', () => {

  it('renders correct NewTopicScreen component', () => {
    const rendered = renderer.create(
      <NewTopicScreen 
        isAddingNewTopic={false}
        addNewTopic={() => {}}
        navigation={{
          setParams: () => {},
          state: {}
        }}
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  })

  it('renders correct HomeScreen component while submiting topic', () => {
    const rendered = renderer.create(
      <NewTopicScreen 
        isAddingNewTopic={true}
        addNewTopic={() => {}}
        navigation={{
          setParams: () => {},
          state: {}
        }}
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  })
})

