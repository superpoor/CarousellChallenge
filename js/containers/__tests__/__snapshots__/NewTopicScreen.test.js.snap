// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NewTopicScreen component renders correct HomeScreen component while submiting topic 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Test NewTopicScreen component renders correct NewTopicScreen component 1`] = `
<View
  keyboardVerticalOffset={64}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Type your topic here"
    style={
      Object {
        "color": "#545454",
        "flex": 1,
        "fontSize": 16,
        "marginVertical": 20,
        "padding": 20,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#747474",
        "fontSize": 14,
        "paddingBottom": 10,
        "paddingRight": 10,
        "textAlign": "right",
      }
    }
  >
    255
     characters left
  </Text>
</View>
`;
